### DISCUSSION TITLE
Writing a saver plugin

### DISCUSSION THREAD

**Discussion by abnlsn** [TiddlyWiki/TiddlyWiki5#5985] [General]
Hi, I am trying to figure out how to write a saver plugin, and am having a really hard time. Could someone point me towards a simple example or something? Thank you so much!

**Comment by saqimtiaz** [comment]
Savers are implemented by modules of the type `saver`.

These are all the savers that ship with the core:
https://github.com/Jermolene/TiddlyWiki5/tree/master/core/modules/savers

Also see the documentation at:
https://tiddlywiki.com/dev/#Saver

**Comment by linonetwo** [comment]
You might need to look at the code of other saver plugins.

At the time I wrote https://github.com/linonetwo/solid-tiddlywiki-syncadaptor  , I truly read much code, to get myself to understand those undocumented life cycle API...

**Comment by dubiouscript** [comment]
imho some scriptable server was (sometimes)helpfull in figuring out what/how was required to save 
... but what scripting lang/methods the *your* server depends on what lang your more accustmed to..

for me (a shell script fan) this : https://github.com/TooTallNate/bashttpd :+1: is.gd 
... it uses netcat or socat to "pipe" requests to a 'server'-program  in my case thats a bash script , but could be any program/language 

see this comment : https://github.com/Jermolene/TiddlyWiki5/issues/3946#issuecomment-493716760
wrt saver methods 

so basicly pick one of 
- saver-handler = "file-based save. That means, it saves 1 tiddlywiki.html file"
and 
- sync adaptors = "save 1 tiddler as xx.tid file to a tiddlers subdirectory"

browser dev tools console / network are probably usefull 

all though admitedly its a broad subject area 

from 
browsers: markup,scripting.xhhtp requests, 
to 
"servers": protocalls ,response, headers , 
...not to mention the server host OS enviroment :scream_cat: 

things *any* one might find usefull to pick apart the tangled techno web 

https://learnxinyminutes.com/
https://learnxinyminutes.com/docs/html/
https://learnxinyminutes.com/docs/javascript/

https://github.com/for-GET/know-your-http-well :fireworks: 
https://github.com/for-GET/know-your-http-well/blob/master/headers.md
https://github.com/for-GET/know-your-http-well/blob/master/methods.md

>  trying to figure out how to write a saver plugin

what is the plugin for ...
- a server you have writen/setup on your system
- some web service 
??..
